### ISSUE TITLE
imagery view doesn't work with new telemetry API

### ISSUE THREAD

**Issue by larkin** [nasa/openmct#1591]
The Imagery view doesn't work with new-style providers.  Short term, the ImageViewPolicy should be updated to read new-style telemetry for checking applicability, and this would resolve the issue.

The example imagery plugin should be updated to expose this bug-- this has been done in the imagery-updates branch.  That can be used as a next step in fixing this.

**Comment by VWoeltjen** [comment]
This will be a significant regression in WARP once transition to new-style providers is complete; marking critical accordingly.

**Comment by larkin** [comment]
I have a fix almost done, I have just one point to run by you @vwoeltjen

Currently, the image view applies to objects with a telemetry value format of 'image'. I would like to change this to a hint of image. That would be more semantic, while allowing custom formatters to be used (say, if the raw values weren't urls, but values that had to be concatenated with a string, or images in blobs that would need to be converted to a dataurl for display). 

It seems proper to use hints to signal the intent / usage of a value, and formats to handle how to interpret that value properly. 

Your thoughts?

**Comment by VWoeltjen** [comment]
__TL;DR:__ Agree with not using a formatter. Don't think `hints` is the best fit; propose new API, but fine with using `hints` if new API is not worth the trouble.

-----

I do think it makes sense that a formatter should not be used for this purpose. `parse` and `validate` are not necessary here, but are definitely necessary for formatters; they have different roles and use cases. (The user isn't going to draw an image and expect back a URL.)

That said, `hints` are fairly vague; I'd rather reserve them for stuff that really doesn't make sense in value metadata. (Otherwise, `hints` just becomes value metadata that's nested in another object.) I also don't see how you'd get from the hint to the code you'd need to run to handle the use cases you've described (isn't the view responsible for interpreting the hint?)

As an alternative, we could use a `conversions` block like:

```
format: "string",
conversions: {
    "imageUrl": "identity" // Or "blobToDataUrl", etc
},
hints: { domain: 1, etc... }
```

...elsewhere...

```
openmct.telemetry.addConversion("identity", function (v) { return v; });
```

...and in the view...

```
openmct.telemetry.convert(metadata.conversions.imageUrl, value);
```

There's a little bit of redundancy there still between conversions and formatters which could be ironed out by splitting formatters into conversions and parsers. It would reduce the number of interfaces (a parser could just be a function that throws errors for validation), and if you have custom string formatting but never want to parse you can just use a conversion and implement less. That said, it's also a bigger change, and I do like that formats are relatively straight and to the point.

All that having been said, I can understand (and share) an aversion to adding to the public API at this time, and a hint would be a smaller change in that regard. I think that's a fine way to go, particularly if the interpretation of those hints can be more succinct and loosely-coupled than what I'm picturing.

**Comment by larkin** [comment]
tl;dr: I agree this is not perfect; I think we still haven't collected enough use cases to accurately represent the design space, and I'm inclined to stick with `hints` as it gives us one place to collect use cases for the time being, and allows us to continue exploring this space as we go, and we can work out a better API in a "Space Balls 2, the search for more money."

-----

I see the following questions views ask of the open mct APIs:

1. Does this telemetry object provide the types of data I'm interested in?
2. How do I get telemetry for this object?
3. How do I format the telemetry for this object for display to a human? 
4. How do I parse it for programmatic consumption? (not all views care for all telemetry values)

* Hints are intended to solve the first problem by attaching metadata that describes the type of telemetry an object provides for views.  
* request/subscribe solve the second problem.
* Formats handle case 3 and 4.  

Hints do place some expectations on Formats-- if a value is hinted as domain or range, it is expected to parse to a numeric value.  If it is hinted as an imageUrl, it is expected to format to an imageUrl, and parse to a ???.  So there is some intended coupling here.

I agree it's a bit confusing w.r.t. imageurls.  

The other problem in my headspace is "how do we prevent plots from displaying on message views?"  Currently, the plot policy is "if it has a number format, then display plot".  I'm thinking that will change to "if it has a domain and range, then display plot".  

Table views will apply to any telemetry object.  But imagery views would need to be higher priority.

I also see hints being used for summary views-- e.g. hints could indicate that a field is useful for grouping, e.g. an object representing incoming packets could support grouping by APID, and could then display total counts of the values with `range` hints.
